### ISSUE TITLE
Add enum for GraphBuilder::_conflicting_references

### ISSUE THREAD

**Issue by jgsogo** [conan-io/conan#3970]
After #3055 is merged, function `_conflicting_references` must return an enumerated value (or named variables).

_Originally posted by @jgsogo in https://github.com/conan-io/conan/pull/3055_

**Comment by jgsogo** [comment]
It's been already done at some point in the past.
